core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,251 @@
+
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotStorageLogsChunk>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+
+    tracing::info!(
+        "Finished storing storage logs chunk {}/{chunks_count}, output stored in {output_filepath}",
+        chunk_id + 1,
+    );
+    Ok(output_filepath)
+}
+
+async fn process_factory_deps(
+    blob_store: &dyn ObjectStore,
+    pool: &ConnectionPool,
+    miniblock_number: MiniblockNumber,
+    l1_batch_number: L1BatchNumber,
+) -> anyhow::Result<String> {
+    tracing::info!("Processing factory dependencies");
+    let mut conn = pool.access_storage().await?;
+    let factory_deps = conn
+        .snapshots_creator_dal()
+        .get_all_factory_deps(miniblock_number)
+        .await;
+    let factory_deps = SnapshotFactoryDependencies { factory_deps };
+    let filename = blob_store
+        .put(l1_batch_number, &factory_deps)
+        .await
+        .context("Error storing factory deps in blob store")?;
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotFactoryDependencies>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+    tracing::info!(
+        "Finished processing factory dependencies, output stored in {}",
+        output_filepath
+    );
+    Ok(output_filepath)
+}
+
+async fn run(blob_store: Box<dyn ObjectStore>, pool: ConnectionPool) -> anyhow::Result<()> {
+    let mut conn = pool.access_storage().await?;
+    let start_time = seconds_since_epoch();
+
+    let l1_batch_number = conn.blocks_dal().get_sealed_l1_batch_number().await? - 1; // we subtract 1 so that after restore, EN node has at least one l1 batch to fetch
+
+    if conn
+        .snapshots_dal()
+        .get_snapshot_metadata(l1_batch_number)
+        .await?
+        .is_some()
+    {
+        tracing::info!(
+            "Snapshot for L1 batch number {} already exists, exiting",
+            l1_batch_number
+        );
+        return Ok(());
+    }
+
+    let miniblock_number = conn
+        .blocks_dal()
+        .get_miniblock_range_of_l1_batch(l1_batch_number)
+        .await?
+        .unwrap()
+        .1;
+    let storage_logs_count = conn
+        .snapshots_creator_dal()
+        .get_storage_logs_count(l1_batch_number)
+        .await?;
+
+    drop(conn);
+
+    //TODO load this from config
+    let chunk_size = 1_000_000;
+    let chunks_count = ceil_div(storage_logs_count, chunk_size);
+
+    tracing::info!(
+        "Creating snapshot for storage logs up to miniblock {}, l1_batch {}",
+        miniblock_number,
+        l1_batch_number.0
+    );
+    tracing::info!(
+        "Starting to generate {} chunks of max size {}",
+        chunks_count,
+        chunk_size
+    );
+
+    let factory_deps_output_file =
+        process_factory_deps(&*blob_store, &pool, miniblock_number, l1_batch_number).await?;
+
+    let mut storage_logs_output_files = vec![];
```

Comment:
Nit: You can specify the correct capacity for the created `Vec`: `Vec::with_capacity(chunks_count)`.

Reply:
Does it have benefits other than better performance? I find the code a bit less readable with the suggested change

Reply:
Performance is the main benefit. TBH, I don't see how `Vec::with_capacity(chunks_count)` is less readable than `vec![]`, but it's personal.